keps/sig-node/5309-kep-template/README.md

```diff
@@ -0,0 +1,875 @@
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+There are many scenarios when a Pod consist of a main task and a sidecar
+container responsible for this task orchestration. The sidecar in this scenario
+receives a signal indicating whether the task is still needed. Allowing the
+orchestration sidecar to terminate or restart the main task would allow fast
+reaction on this signal.
+
+The KEP introduces the communication channel between kubelet and a container in
+a Pod, which may be extended to a lot of other scenarios in future.
+
+## Motivation
+
+Kubernetes workloads today rely on the API Server for pod creation, deletion,
+and lifecycle management. However, certain advanced use cases require
+self-managed, dynamic pod orchestration within a node while minimizing direct
+API Server interactions. This KEP proposes Self-Orchestrating Pods (SOPs), a
+mechanism that allows a pod to create, manage, and terminate its
+containers without requiring constant communication with the Kubernetes control
+plane or escalating privileges to get access to CRI API.
+
+This capability is particularly useful in resource-intensive environments where
+high-frequency pod management operations could lead to excessive API Server
+load and unnecessary delays in effective work, stalling the expensive resources.
+For instance, in GPU workloads, features like NVIDIA IMEX (Inter-GPU
+Memory Exchange) require auxiliary daemon processes running alongside workloads.
+Instead of relying on external DaemonSets or the API Server for deployment, a
+Self-Orchestrating Pod would dynamically spawn and manage its own supporting
+containers based on defined constraints.
+
+By extending this concept beyond GPU and DRA-specific implementations,
+Self-Orchestrating Pods could benefit other high-performance computing (HPC),
+real-time systems, edge computing, and AI/ML workloads that require fine-grained
+control over their execution environment.
+
+### Key Features & Benefits
+
+- Reduced API Server Load: Workloads manage their own supporting containers
+  without frequent API Server interactions.
+- Fine-Grained Workload Control: Pods can create and terminate sub-containers
```

Comment:
The usecases below don't talk about creating - rather about terminating and restarting containers.

Creation becomes much more tricky I think - do we need to include it?